onnxscript/tests/function_libs/torch_lib/ops_test_data.py

```diff
@@ -662,6 +662,10 @@ def _where_input_wrangler(
     TorchLibOpInfo("cosh", core_ops.aten_cosh),
     TorchLibOpInfo("cross", core_ops.aten_cross),
     # TorchLibOpInfo("detach", core_ops.aten_detach),  # detach is not in OP-TEST-DB
+    TorchLibOpInfo("diagonal", core_ops.aten_diagonal, trace_only=True).xfail(
```

Comment:
We can create a BOOL overload for Boolean inputs. We can do this in a separate if desired

Reply:
Added bool overload function.